editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/features/custom/CollapseFeature.java

```diff
@@ -151,16 +165,23 @@ public void collapseShape(PictogramElement pe) {
 		if (rsf.canExecute(context1)) {
 			rsf.execute(context1);
 		}
-	 	 
-		if(!childFiguresVisible) {
-			Graphiti.getPeService().setPropertyValue(pe, PROP_COLLAPSED_STATE, "true");
-		}
-		
-		//visible/invisible all the children
-		makeChildrenInvisible(cs, childFiguresVisible);
 	}
 
 	/**
+	 * updates the border color to reflect the collapse state of the figure
+	 * 
+	 * @param cs
+	 * @param collapsed
+	 */
+	public void updateBorderStyle(ContainerShape cs, boolean collapsed) {
+		IGaService gaService = Graphiti.getGaService();
+		Color col = collapsed ? gaService.manageColor(getDiagram(), StyleUtil.CONTAINER_FIGURE_COLLAPSED_BORDER_COLOR) : gaService.manageColor(getDiagram(), StyleUtil.CONTAINER_FIGURE_BORDER_COLOR); 
+		GraphicsAlgorithm ga = cs.getGraphicsAlgorithm();
+		ga.setLineStyle(collapsed ? LineStyle.DOT : LineStyle.SOLID);
```

Comment:
I prefer with LineStyle.DASH

![image](https://cloud.githubusercontent.com/assets/1105127/19444800/40a88aaa-9492-11e6-9e48-778c51beb32f.png)

than LineStyle.DOT

![image](https://cloud.githubusercontent.com/assets/1105127/19444821/53daddc6-9492-11e6-96df-1782e14ba8ff.png)

What do you think?
DOT are not visible enough for me